src/Renderer/ThreeExtended/Feature2Mesh.js

```diff
@@ -1,90 +1,338 @@
 import * as THREE from 'three';
 import Earcut from 'earcut';
 
-function coordinatesToMesh(coordinates) {
-    if (!coordinates) {
-        return;
+function isFunction(functionToCheck) {
+    return typeof functionToCheck === 'function';
+}
+
+function getAltitude(options, properties) {
+    if (options.altitude) {
+        if (isFunction(options.altitude)) {
+            return options.altitude(properties);
+        } else {
+            return options.altitude;
+        }
     }
-    // create geometry
-    const geometry = new THREE.BufferGeometry();
+    return 0;
+}
+
+function getExtrude(options, properties) {
+    if (options.extrude) {
+        if (isFunction(options.extrude)) {
+            return options.extrude(properties);
+        } else {
+            return options.extrude;
+        }
+    }
+    return 0;
+}
+
+function randomColor() {
+    const randomColor = new THREE.Color();
+    randomColor.setHex(Math.random() * 0xffffff);
+    return randomColor;
+}
+
+function getColor(options, properties) {
+    if (options.color) {
+        if (isFunction(options.color)) {
+            return options.color(properties);
+        } else {
+            return options.color;
+        }
+    }
+    return randomColor();
+}
 
-    const vertices = new Float32Array(3 * coordinates.coordinates.length);
+function createColorArray(length, color, brightness) {
+    const colors = new Array(length * 3);
+    for (let i = 0; i < length; ++i) {
+        colors[3 * i] = color.r * brightness;
+        colors[3 * i + 1] = color.g * brightness;
+        colors[3 * i + 2] = color.b * brightness;
+    }
+    return colors;
+}
+
+/*
+ * Convert coordinates to vertices positionned at a given altitude
+ *
+ * @param  {Coordinate[]} contour - Coordinates of a feature
+ * @param  {number} altitude - Altitude of the feature
+ * @return {Vector3[]} vertices
+ */
+function coordinatesToVertice(contour, altitude) {
+    // position in the vertices result
     let offset = 0;
-    for (const coordinate of coordinates.coordinates) {
-        vertices[offset] = coordinate._values[0];
-        vertices[offset + 1] = coordinate._values[1];
-        vertices[offset + 2] = coordinate._values[2];
+    const vertices = new Array(contour.length * 3);
+    // loop over contour coodinates
+    for (const coordinate of contour) {
+        // convert coordinate to position
+        const vec = coordinate.xyz();
+        // get the normal vector.
+        const normal = coordinate.geodesicNormal;
+        // move the vertex following the normal, to put the point on the good altitude
+        vec.add(normal.clone().multiplyScalar(altitude));
```

Comment:
You're extruding along the normal and I don't think this is good idea: what about extruding along the Z axis then rotating your mesh so their local Z is aligned with the geodesicNormal at their position?

Reply:
No, It's an extrusion local to each points.

Imagine that you want to extrude a bigger polygon on the globe. For example, if we want to extrude a polygon representing region, a country or a continent. In this case, your extrusion on Z won't work.

Reply:
Then, use a rotation that is an average of all the normals, for instance.
And extrude along the diff between the normal at one point and this average.

WDYT?

Reply:
(or the normal at the center, it doesn't really matter,just pick the easiest to compute)

Reply:
Your animation works on planar, because Z axis make sense on planar.

But I don't want to make a wrong approximation to handle this case not adapted at all for globe.

If it's really important for you, I can add the normal vector to the mesh object, to let the user do this kind of scale and add this pretty scale example to the examples.

WDYT ?

Reply:
The thing is: you're building meshes in global coordinates if you extrude them this way... and it's not the way three.js works (ie: the whole matrix/matrixWorld thing is built on one assumption: each element of the scene uses local coordinates which are then transformed using `matrixWorld`)

Reply:
The existing code create meshes in global coordinates.

Your suggestion is interesting, but it's a new feature : put the mesh in local coordinates, adding rotations...

So we'll do that in another PR.

Reply:
> The existing code create meshes in global coordinates.

Not really. They're created in global coords but then vertices position are patched back to local coordinates (see `applyOffset` in FeatureProcessing`).

This wasn't the best technical option, but it was working and didn't cause issue because geometries were flat anyway.

IMHO you just have to compute a matrix that make sense:
- matrix's translation will probably identical to the `offset` from `applyOffset`
- matrix's rotation could be the geodesicNormal at node.center.
Then you can inverse this matrix and apply it to every vertices position you compute.

Reply:
Again, we can do that in another PR.

Reply:
Will you do this other PR? When?

Because I fear that the current implement fits perfectly your use-case and that you'll never bother fixing it (or it'll always be a low-priority item for you)...

Reply:
It's not a fix, it's a new feature.

The feature you speak about "Have the mesh in a local coordinate" was not in iTowns before. 

Adding extrusion to the polygons doesn't necessary mean add this new feature "Have the mesh in a local coordinate".

We can add an issue to speak more about this subject, but it's not the purpose of this PR.

Reply:
I disagree - mainly because I'm pretty sure this "other PR" will never happen.

Anyway I won't block on this specific point anymore since this PR implements a useful feature.

Reply:
Yes indeed it's not the purpose of this PR, furthermore the new feature PR (when, who) is out of scope this code review. HTH

Reply:
I add this issue https://github.com/iTowns/itowns/issues/558